### PR TITLE
[1/x][dynamic child loading] Remove child count

### PR DESCRIPTION
**Pull request by tnowacki**
- In preperation for dynamic child loading, removing child counts
- This will simplify some parts of the migration slightly
- I am not fussing too much over tests as they will change drastically later

### CODE REVIEW

**Review by sblackshear — APPROVED** [review]

File: crates/sui-adapter/src/in_memory_storage.rs [hunk 1]
```diff
@@ -26,7 +26,9 @@ impl BackingPackageStore for InMemoryStorage {
 
 impl ParentSync for InMemoryStorage {
     fn get_latest_parent_entry_ref(&self, object_id: ObjectID) -> SuiResult<Option<ObjectRef>> {
-        debug_assert!(!self.persistent.contains_key(&object_id));
+        if let Some(obj) = self.persistent.get(&object_id) {
```
Comment (lxfind) [inline crates/sui-adapter/src/in_memory_storage.rs:29]: Why was this changed? Was this a bug previously?

Comment (tnowacki) [reply crates/sui-adapter/src/in_memory_storage.rs:29]: Err wrong PR (though you will see that bug soon :) ). So it was probably just an untested codepath
